src/Aspire.Hosting.Azure.AppContainers/ContainerAppExtensions.cs

```diff
@@ -0,0 +1,102 @@
+    /// passing them to the <see cref="ConfigureCustomDomain(ContainerApp, IResourceBuilder{ParameterResource}, IResourceBuilder{ParameterResource})"/>
+    /// method via the <see cref="AzureContainerAppContainerExtensions.PublishAsAzureContainerApp{T}(IResourceBuilder{T}, Action{ResourceModuleConstruct, ContainerApp})"/>
+    /// extension method.
+    /// <code lang="C#">
+    /// var builder = DistributedApplication.CreateBuilder();
+    /// var customDomain = builder.AddParameter("customDomain"); // Value provided at first deployment.
+    /// var certificateName = builder.AddParameter("certificateName"); // Value provided at second and subsequent deployments.
+    /// builder.AddProject&lt;Projects.InventoryService&gt;("inventory")
+    ///        .PublishAsAzureContainerApp((module, app) =>
+    ///        {
+    ///          app.ConfigureCustomDomain(customDomain, certificateName);
+    ///        });
+    /// </code>
+    /// </example>
+    [Experimental("ASPIREACADOMAINS001", UrlFormat = "https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+    public static void ConfigureCustomDomain(this ContainerApp app, IResourceBuilder<ParameterResource> customDomain, IResourceBuilder<ParameterResource> certificateName)
+    {
+        if (app.ParentInfrastructure is not ResourceModuleConstruct module)
+        {
+            throw new ArgumentException("Cannot configure custom domain when resource is not parented by ResourceModuleConstruct.", nameof(app));
+        }
+
+        var containerAppManagedEnvironmentIdParameter = module.GetResources().OfType<ProvisioningParameter>().Single(
+            p => p.IdentifierName == "outputs_azure_container_apps_environment_id");
+        var certificatNameParameter = certificateName.AsProvisioningParameter(module);
+        var customDomainParameter = customDomain.AsProvisioningParameter(module);
+
+        var bindingTypeConditional = new ConditionalExpression(
+            new BinaryExpression(
+                new IdentifierExpression(certificatNameParameter.IdentifierName),
+                BinaryOperator.NotEqual,
+                new StringLiteral(string.Empty)),
+            new StringLiteral("SniEnabled"),
+            new StringLiteral("Disabled")
+            );
+
+        var certificateOrEmpty = new ConditionalExpression(
+            new BinaryExpression(
+                new IdentifierExpression(certificatNameParameter.IdentifierName),
+                BinaryOperator.NotEqual,
+                new StringLiteral(string.Empty)),
+            new InterpolatedString(
+                "{0}/managedCertificates/{1}",
+                [
+                 new IdentifierExpression(containerAppManagedEnvironmentIdParameter.IdentifierName),
+                    new IdentifierExpression(certificatNameParameter.IdentifierName)
+                 ]),
+            new NullLiteral()
+            );
+
+        app.Configuration.Value!.Ingress!.Value!.CustomDomains = new BicepList<ContainerAppCustomDomain>()
```

Comment:
This will erase anything done before. Is that the intent?

Reply:
If we didn't provide anything anything that was added manually would be removed. By doing this we allow folks to specify it as part of the deployment.

Reply:
... and if someone is customizing those properties they probably aren't using this method anyway they are doing something custom.